### DISCUSSION TITLE
Promotion

### DISCUSSION THREAD

**Discussion by syssam** [netsuitephp/netsuite-php#142] [Q&A]
Any one know how to apply promotion to cash sale and sales order

**Comment by jrebs** [comment]
Not sure exactly what you mean by promotion, but if you just mean how to apply a discount to the order, netsuite has a concept of discounts on the sales order and cash sales records.

**Reply by syssam** [reply]
Thanks for your reply. here is the promotion document, https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1519050978.html
I could not find any related API to do that.

**Reply by syssam** [reply]
In the NetSuite UI, there has a section to apply promotion code or auto-apply promotion.

**Comment by neclimdul** [comment]
Never interacted with this part of netsuite myself but there's a good bit to do with promotions in the library.

```
src/Classes/Campaign.php:    public $promotionCode;
src/Classes/Campaign.php:        "promotionCode" => "RecordRef",
src/Classes/CampaignSearch.php:    public $promotionCodeJoin;
src/Classes/CampaignSearch.php:        "promotionCodeJoin" => "PromotionCodeSearchBasic",
src/Classes/CampaignSearchRow.php:    public $promotionCodeJoin;
src/Classes/CampaignSearchRow.php:        "promotionCodeJoin" => "PromotionCodeSearchRowBasic",
src/Classes/CashSale.php:    public $promotionsList;
src/Classes/CashSale.php:        "promotionsList" => "PromotionsList",
src/Classes/CouponCode.php:    public $promotion;
src/Classes/CouponCode.php:        "promotion" => "RecordRef",
src/Classes/CouponCodeSearchBasic.php:    public $promotion;
src/Classes/CouponCodeSearchBasic.php:        "promotion" => "SearchMultiSelectField",
src/Classes/CouponCodeSearchRowBasic.php:    public $promotion;
src/Classes/CouponCodeSearchRowBasic.php:        "promotion" => "SearchColumnStringField[]",
src/Classes/CustomSale.php:    public $promotionsList;
src/Classes/CustomSale.php:        "promotionsList" => "PromotionsList",
src/Classes/DeletedRecordType.php:    const promotionCode = "promotionCode";
src/Classes/Estimate.php:    public $promotionsList;
src/Classes/Estimate.php:        "promotionsList" => "PromotionsList",
src/Classes/Invoice.php:    public $promotionsList;
src/Classes/Invoice.php:        "promotionsList" => "PromotionsList",
src/Classes/PermissionCode.php:    const _promotionCode = "_promotionCode";
src/Classes/PromotionsList.php:    public $promotions;
src/Classes/PromotionsList.php:        "promotions" => "Promotions[]",
src/Classes/RecordType.php:    const promotionCode = "promotionCode";
src/Classes/SalesOrder.php:    public $promotionsList;
src/Classes/SalesOrder.php:        "promotionsList" => "PromotionsList",
src/Classes/SearchRecordType.php:    const promotionCode = "promotionCode";
```

Skimming, my guess would be that you'd set the sales order property with the promotions.

```
$order = new SalesOrder();
// Populate oorder as needed
$promo = new PermissionCode();
// Populate promotion as needed
$order->promotionsList = new PromotionsList();
$order->promotionsList->promotions[] = $promo;
// Save or what ever.
```

**Comment by syssam** [comment]
![image](https://user-images.githubusercontent.com/20774834/210690777-e792acb1-5786-463a-a0d7-2e8a6eeaa4a0.png)
Yup, I know the promotion list parameter, however, it is not working.
I apply for promotion on the Netsuite UI and dump the struct, you will see the value is null

**Comment by syssam** [comment]
Sorry, I make a mistake, promotion list has a value, however, It does not work on free gift promotion

**Comment by syssam** [comment]
![image](https://user-images.githubusercontent.com/20774834/210695227-ff08c268-adef-4b4e-aeef-b6e666682ce7.png)
![image](https://user-images.githubusercontent.com/20774834/210695344-ba6c7101-e614-413a-af6f-e10c242dbe30.png)
Items need FREE GIFT PROMOTION parameter, however I could not found that...
